litgpt/pretrain.py

```diff
@@ -169,12 +173,13 @@ def main(
 
     model = torch.compile(model)
     model = fabric.setup(model)
+
     optimizer = torch.optim.AdamW(
         model.parameters(),
         lr=train.learning_rate,
         weight_decay=train.weight_decay,
         betas=(train.beta1, train.beta2),
-        fused=True,
+        fused=torch.cuda.is_available(),
```

Comment:
Is this robust enough or are there cases where this could be problematic?

Reply:
This would be more reliable, in case it's run on CPU in a machine with a GPU

```suggestion
        fused=fabric.device.type == "cuda",
```